Fixed Legend position option in the Line graph > slope option

derekagorhom opened the pull request:
Fixes #9196 
@rachelkg , @N-thony , @rdstern I have disabled the lower legend position checkbox when the user is in the Slope option of the line graph dialogue. I have added a drop down input position to the legend option at the top. 
Can you review this PR
Thanks

rdstern commented:
@derekagorhom I hope the solution is simpler, namely delete the top legend control.  

We added the lower control to many graph dialogs, and I assume we did not notice it has an inferior version there.  Can you try that anyway.

derekagorhom commented:
@rdstern I have made so that the lower legend option work for the slope option now and deleted the top option...can you have a look
Thanks

rdstern reviewed (changes requested):
@derekagorhom can you also fix the single Legend control.  

By the way, the greenrice data is in the agritutorial package if you want to use the same data as in the issue. 
Start by making W and N into factors.  You could use the slope (as Rachel does) or start with the line plot - the graph is silly there, but that's not the point. 
Start with the dialog as so:

![image](https://github.com/user-attachments/assets/652dfc6d-030d-4bd5-a4e2-58c7b2074d8f)

Notice I haven't done anything to the legend, so it is unchecked.

Press Ok.  Then get the dialog back and the legend control is checked.  That shouldn't happen.  Note I checked with the latest version and the control is ok there.  So, this is an example where you have fixed one fault and now introduced a new problem in that dialog, that wasn't there before.  And this new problem affects all the 5 options at the top of the dialog.  Please fix, and maybe use this example of a "regression in the code" to check that you don't do this sort of thing too often.

derekagorhom commented:
> @derekagorhom can you also fix the single Legend control.
> 
> By the way, the greenrice data is in the agritutorial package if you want to use the same data as in the issue. Start by making W and N into factors. You could use the slope (as Rachel does) or start with the line plot - the graph is silly there, but that's not the point. Start with the dialog as so:
> 
> ![image](https://private-user-images.githubusercontent.com/11226469/379841998-652dfc6d-030d-4bd5-a4e2-58c7b2074d8f.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzAxMDM2MDcsIm5iZiI6MTczMDEwMzMwNywicGF0aCI6Ii8xMTIyNjQ2OS8zNzk4NDE5OTgtNjUyZGZjNmQtMDMwZC00YmQ1LWE0ZTItNThjN2IyMDc0ZDhmLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDEwMjglMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMDI4VDA4MTUwN1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTQzY2ExYTQ4ZjQyYzhiODJiMzIyNDllNzk0Y2UwZGY0MTU3ZjI5ZWYxMGM0Y2JiYzBhMDJkY2RiYTczMDAwNDEmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.Dg3dfWtzz3evkU1LcBsxEGCaGiv26IQe41ckDOJ6418)
> 
> Notice I haven't done anything to the legend, so it is unchecked.
> 
> Press Ok. Then get the dialog back and the legend control is checked. That shouldn't happen. Note I checked with the latest version and the control is ok there. So, this is an example where you have fixed one fault and now introduced a new problem in that dialog, that wasn't there before. And this new problem affects all the 5 options at the top of the dialog. Please fix, and maybe use this example of a "regression in the code" to check that you don't do this sort of thing too often.

@rdstern I have fixed the problem you identified. 
can you have a look again 
Thanks

rdstern reviewed (approved):
Thanks @derekagorhom that looks fine now.  @N-thony over to you

rdstern commented:
@N-thony over to you

derekagorhom commented on instat/dlgLinePlot.vb:
```diff
@@ -858,9 +855,22 @@ Public Class dlgLinePlot
+        End If
+        If rdoLine.Checked AndAlso rdoLinerange.Checked AndAlso rdoSmoothing.Checked AndAlso rdoDumbbell.Checked Then
+            AddRemoveTheme()
+        End If
```
I changed it from `AndAlso` to `OrElse`

derekagorhom replied:
No... these are the other options in the dialog

derekagorhom commented on instat/dlgLinePlot.vb:
```diff
@@ -1278,4 +1272,8 @@ Public Class dlgLinePlot
+
+    Private Sub AllControl_ControlContentsChanged(ucrChangedControl As ucrCore) Handles ucrVariablesAsFactorForLinePlot.ControlContentsChanged, ucrSave.ControlContentsChanged, ucrReceiverYMin.ControlContentsChanged, ucrReceiverYMax.ControlContentsChanged, ucrReceiverXEnd.ControlContentsChanged, ucrReceiverX.ControlContentsChanged, ucrReceiverSlopeY.ControlContentsChanged, ucrReceiverSlopeX.Contro ...
+
+    End Sub
```
I have removed this sub, i think its from me making some changes in the designer

derekagorhom commented:
@rdstern can review this again

rdstern reviewed (approved):
@N-thony back to you
